### PR TITLE
(feat) Add ability to filter observations in the generic patient widget app by encounter type

### PR DESCRIPTION
**Pull request by icrc-agomes**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [x] My work includes tests or is validated by existing tests.

## Summary
In `esm-generic-patient-widgets-app`, it was needed to implement the filter of observations endpoint by encounter type. Some changes were also made in the mock data for testing.

## Related Issue
This branch is a continuation of this pull request [#827](https://github.com/openmrs/openmrs-esm-patient-chart/pull/827)

### CODE REVIEW

File: packages/esm-generic-patient-widgets-app/src/resources/useObs.ts [hunk 1]
```diff
@@ -4,7 +4,9 @@ import { openmrsFetch, fhirBaseUrl, useConfig, FHIRResource, FHIRCode } from '@o
 export const pageSize = 100;
 
 export function useObs(patientUuid: string): UseObsResult {
-  const { data } = useConfig();
+  const { encounterTypes, data } = useConfig();
+
+  const UrlEncounterTypes: string = encounterTypes.length ? `&encounter.type=${encounterTypes.toString()}` : '';
```
Comment (denniskigen) [inline packages/esm-generic-patient-widgets-app/src/resources/useObs.ts:9]:
```suggestion
  const urlEncounterTypes: string = encounterTypes.length ? `&encounter.type=${encounterTypes.toString()}` : '';
```

File: packages/esm-generic-patient-widgets-app/src/resources/useObs.ts [hunk 2]
```diff
@@ -14,8 +16,8 @@ export function useObs(patientUuid: string): UseObsResult {
     `${fhirBaseUrl}/Observation?subject:Patient=${patientUuid}&code=` +
       data.map((d) => d.concept).join(',') +
       '&_summary=data&_sort=-date' +
-      `&_count=${pageSize}
-  `,
+      `&_count=${pageSize}` +
+      UrlEncounterTypes,
```
Comment (denniskigen) [inline packages/esm-generic-patient-widgets-app/src/resources/useObs.ts:20]:
```suggestion
      urlEncounterTypes,
```

File: packages/esm-generic-patient-widgets-app/src/config-schema.ts [hunk 3]
```diff
@@ -47,6 +47,18 @@ export const configSchema = {
       _default: 5,
     },
   },
+  showGraphByDefault: {
+    _type: Type.Boolean,
+    _description: 'Displayed graph by default',
+    _default: true,
+  },
+  encounterTypes: {
```
Comment (denniskigen) [inline packages/esm-generic-patient-widgets-app/src/config-schema.ts:55]: We should have a `_description` for this property as well.

File: packages/esm-generic-patient-widgets-app/src/resources/useObs.ts (hunk 1 above)
Comment (denniskigen) [reply packages/esm-generic-patient-widgets-app/src/resources/useObs.ts:9]: Seems better to name this variable `encounterTypes`.

**Review by denniskigen — COMMENTED** [review]
Generally looks good. I've left a few comments to address then this should be good to go.

File: packages/esm-generic-patient-widgets-app/src/resources/useObs.ts (hunk 1 above)
Comment (icrc-agomes) [reply packages/esm-generic-patient-widgets-app/src/resources/useObs.ts]: I used the suggested change 'urlEncounterTypes'

File: packages/esm-generic-patient-widgets-app/src/config-schema.ts (hunk 3 above)
Comment (icrc-agomes) [reply packages/esm-generic-patient-widgets-app/src/config-schema.ts:55]: Parameter added

**Review by hadijahkyampeire — APPROVED** [review]
Looks Good, thanks @icrc-agomes 👍

**Comment by icrc-agomes** [comment]
> Looks Good, thanks @icrc-agomes 👍 , except for the merge conflicts, otherwise, we can merge it.

@hadijahkyampeire conflicts fixed :) Thanks.

**Pull request merged**